dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopScripts.java

```diff
@@ -126,10 +128,20 @@ static Path create(String scriptFilename, byte[] scriptBody) throws IOException
     LOG.info("Extracting '{}' to '{}'...", scriptFilename, scriptDir);
     Path scriptPath = scriptDir.resolve(scriptFilename);
     Files.write(scriptPath, scriptBody);
+    sanityCheckFileContents(scriptPath, scriptBody);
     scriptPath.toFile().setExecutable(true);
     return scriptPath;
   }
 
+  private static void sanityCheckFileContents(Path scriptPath, byte[] expectedScriptBody)
+      throws IOException {
+    byte[] scriptBody = Files.readAllBytes(scriptPath);
+    Preconditions.checkState(
+        Arrays.equals(expectedScriptBody, scriptBody),
+        "Sanity check failed: File contents doesn't match the script body. Path='%s'.",
```

Comment:
What are some potential causes for this contents mismatch?

Reply:
Hard to tell. Some unexpected behavior, where writing would succeed, but reading would yield a different result. It's a sanity check to make sure that the script being executed contains the expected body. The scripts are very small, so it should not affect the performance.

Reply:
I expect that if `Files.write` finishes without an exception, then the file has been written correctly and the check is not needed.

Unless we can create a reasonable unit test case that would trigger this scenario, I suggest that this check is not needed.